controller/backup_controller.go

```diff
@@ -307,6 +307,23 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 			bc.eventRecorder.Eventf(backup, corev1.EventTypeWarning, string(backup.Status.State), "Failed backup %s has been deleted: %s", backup.Name, backup.Status.Error)
 		}
 
+		// do the best effort to delete the snapshot
+		snapshot, err := bc.ds.GetSnapshot(backup.Spec.SnapshotName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				logrus.WithError(err).WithFields(logrus.Fields{
+					"backup":   backup.Name,
+					"snapshot": snapshot.Name,
+				}).Warn("Failed to get snapshot")
+			}
+			return nil
+		}
+		if err = bc.ds.DeleteSnapshot(snapshot.Name); err != nil {
```

Comment:
This is to trigger the snapshot deletion asynchronously, so when the backup is deleted, it doesn't mean the snapshot has been deleted, but it will eventually be deleted. We need to describe this clearly in doc.